woocommerce/woocommerce-android
# Privacy Choices: Handle Save CTA

Closes: #8948


### Description


This PR adds support for taping the `Save` button on the Privacy Banner.

### Testing instructions


Hint: you can use the new option in `Developer Options` to trigger showing the Privacy Banner again. See here: p1684926805528519-slack-C02KUCFCSFP

#### TC1 WPCOM account

1. Log in as WPCOM user
2. On the Privacy Banner, change the "Analytics" switch
3. Tap "Save"
4. Assert that after a second, the Privacy Banner is dismissed
5. Go to https://wordpress.com/me/privacy and check if the Privacy Settings were updated
6. Restart the app
7. Assert that Privacy Banner is not shown again

#### TC2 Non-WPCOM account

1. Log in as non-WPCOM user
2. On the Privacy Banner, change the "Analytics" switch
3. Tap "Save"
4. Assert that the Privacy Banner is dismissed
5. Assert that Tracking is disabled (no "Tracked :" logs)
6. Restart the app
7. Assert that the Privacy Banner is not shown again

#### TC3 WPCOM account but error

1. Log in as WPCOM user
2. Turn off the internet connection
3. On the Privacy Banner, change the "Analytics" switch
4. Tap "Save"
5. Assert that the Privacy Banner is dismissed and Snackbar with "Retry" button is shown
6. Turn on the internet connection
7. Tap on "Retry"
8. Assert that Snack bar is hidden
9. Go to https://wordpress.com/me/privacy and check if the Privacy Settings were updated


### Images/gif


#### Success

https://github.com/woocommerce/woocommerce-android/assets/5845095/a1877fc4-ea89-4184-9126-240bd7c2b9d9

#### Failure (and then success)

https://github.com/woocommerce/woocommerce-android/assets/5845095/e441632c-4818-4a11-bfcb-70e918ef1814
